### PR TITLE
Added ability to define ports mapping at the service level. (fixes #165)

### LINKED ISSUE #165: Ability to define ports mapping at the service level
Currently maestro only supports ports mapping defined in service 'instance' level.

http://maestro-ng.readthedocs.org/en/latest/port_mapping.html

However, when the ports mapping is the same for each instance of the service, it would be nice to define ports mapping at service level. We did refactor the ports mapping to service level and did not receive error when maestro parsed the yaml. It took us a while to notice that ports mapping only apply to instance level.

```
  zookeeper:
    image: docker.cenx.localnet:5000/zookeeper
    lifecycle:
      running: [{type: tcp, port: client}]
    ports: {client: 2181, peer: 2888, election: 3888, jmx: 9000}
    instances:
      zk1:
        ship: ship1
      zk2:
        ship: ship2
      zk3:
        ship: ship3
```

versus

```
  zookeeper:
    image: docker.cenx.localnet:5000/zookeeper
    lifecycle:
      running: [{type: tcp, port: client}]
    instances:
      zk1:
        ship: ship1
        ports: {client: 2181, peer: 2888, election: 3888, jmx: 9000}
      zk2:
        ship: ship2
        ports: {client: 2181, peer: 2888, election: 3888, jmx: 9000}
      zk3:
        ship: ship3
        ports: {client: 2181, peer: 2888, election: 3888, jmx: 9000}
```

### PR DESCRIPTION
**Pull request by zsuzhengdu**
Fixes #165

### CODE REVIEW

**Comment by zsuzhengdu** [comment]
@mpetazzoni  Would you please spare some of your precious time to review this PR? Thanks!

File: maestro/entities.py [hunk 1]
```diff
@@ -224,6 +226,7 @@ def __init__(self, name, image, omit=True, schema=None, env=None,
         })
         self._lifecycle = lifecycle or {}
         self._limits = limits or {}
+        self.ports = ports or {}
```
Comment (mpetazzoni) [inline maestro/entities.py]: Should be `self._ports`

File: tests/unittests.py [hunk 2]
```diff
@@ -51,6 +51,16 @@ def test_limits_option(self):
...
+    def test_no_ports_option(self):
+        service = entities.Service('foo', 'stackbrew/ubuntu:13.10')
+        self.assertEqual(service.ports, {})
+
+    def test_ports_option(self):
+        service = entities.Service('foo',
+                                   'stackbrew/ubuntu:13.10',
+                                   ports={'server: 4848'})
+        self.assertEqual(service.ports, {'server: 4848'})
+
```
Comment (mpetazzoni) [inline tests/unittests.py]: Add a test for containers to check that ports defined at the service level correctly propagate down to the container instances' ports.

File: maestro/entities.py (hunk 1 above)
Comment (zsuzhengdu) [inline maestro/entities.py]: Updated as requested.

File: tests/unittests.py (hunk 2 above)
Comment (zsuzhengdu) [inline tests/unittests.py]: Added test_ports_propagates_from_service in ContainerTest.

**Pull request merged**